lib/genome/online_updaters/docm/updater.rb

```diff
@@ -0,0 +1,132 @@
+require 'net/http'
+module Genome; module OnlineUpdaters; module Docm
+  class Updater
+    attr_reader :new_version
+    def initialize(source_db_version = Date.today.strftime("%d-%B-%Y"))
+      @new_version = source_db_version
+    end
+
+    def update
+      remove_existing_source
+      source = create_new_source
+      create_interaction_claims(source)
+    end
+
+    private
+    def create_interaction_claims(source)
+      api_client = ApiClient.new
+      api_client.variants.each do |variant|
+        interaction_information = parse_interaction_information(variant)
+        interaction_information.each do |interaction_info|
+          gc = get_or_create_gene_claim(variant, source)
+          dc = get_or_create_drug_claim(interaction_info, source)
+          get_or_create_interaction_claim(gc, dc, interaction_info, variant['diseases'], source)
+        end
+      end
+    end
+
+    def get_or_create_gene_claim(variant, source)
+      DataModel::GeneClaim.where(
+        source_id: source.id,
+        nomenclature: 'DoCM Entrez Gene Symbol',
+        name: variant['gene']
+      ).first_or_create
+      #Entrez Gene Id
+    end
+
+    def parse_interaction_information(variant)
+      if !variant.key?('meta')
+        return []
+      end
+      fields = variant['meta'].first['Drug Interaction Data']['fields']
```

Comment:
For a little bit of future proofing, there's no guarantee that there's only going to be one table in the`meta` section or that the drug interaction data one will be the first one, so it might be safer to do something like:

```
return [] unless variant.include?('meta')
drug_data = variant['meta'].find { |table| table.include?('Drug Interaction Data') }
return [] unless drug_data.present?
fields = drug_data['fields']
rows = drug_data['rows']
```